### PR TITLE
What would a WeChat replacement need?

### PR DESCRIPTION
**Pull request by oskarth**
https://github.com/vacp2p/research/issues/21

Will post tomorrow or so to spread posts out and avoid Easter.

### CODE REVIEW

File: _posts/2020-04-16-wechat-replacement-need.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+## What's next?
+
+We've look at what WeChat provides and what we'd like an alternative to look like. We've also seen a few principal modules that are necessary to achieve those goals. To achieve all of this is a daunting task, and one might call it overly ambitiuous. We've also seen how far we've come with some of the goals, and how a lot of the pieces are there, in one form or another. Then it is a question of putting them all together in the right mix.
+
+The good news is that a lot of people are working all these building blocks and thinking about these problems. Compared to a few years ago we've come quite far when it comes to p2p infrastructure, privacy, security, scalability, and general developer mass and mindshare. If you want to join us in building some of these building blocks, and assembling them, check out our forum.
+
+PS. We are hiring. DS
+
+**TODO: Insert link to JD**
```
Comment (oskarth) [inline _posts/2020-04-16-wechat-replacement-need.md:244]: Don't forget

File: _posts/2020-04-16-wechat-replacement-need.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+### Extensible mini apps
+
+This relates to the compute and storage module above. Essentially we want to provide mini apps as in WeChat, but to do so in a way that is compatible with what we want to achieve more generally. This allows individuals and small businesses to create small tools for various purposes, and coordinate with strangers. E.g. booking a cab or getting an insurance, and so on.
+
+This has a higher dependency on the contracting/general computation aspect. I.e. often it isn't only a transaction, but you might want to encode some specific rules here that strangers can abide by without having too high trust requirements. As a simple example: escrows.
+
+This also needs an open API that anyone can use. It should be properly secured, so using one doesn't compromise the rest of the system it is operating in. To be censorship resistant it requires the routing and storage component to work properly.
+
+## Where are we now?
```
Comment (oskarth) [inline _posts/2020-04-16-wechat-replacement-need.md:204]: FYI @arnetheduck I added this rough section

File: _posts/2020-04-16-wechat-replacement-need.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+**Private and secure communication**. Signal has pioneered a lot of this, following OTR. Double Ratchet, X3DH. E2EE is minimum these days, and properties like PFS and PD are getting better. For metadata protection, you have Tor, with its faults, and more active research on mixnets and private information retrieval, etc.
+
+**Censorship-resistance**. This covers a lot of ground across the spectrum. You have technologies like Bittorrent, Bitcoin/Ethereum, Tor obfuscated transports, E2EE by default, partial mesh networks in production, abilit to move/replicate host machines more quickly have all made this more of a reality than it used to be. this easier. Of course, techniques such as deep packet inspection and internet shutdowns have increased.
+
+**Decentralization**. Cryptocurrencies, projects like libp2p and IPFS. Need to be mindful here of many projects that claim decentralization but are still vulnerable to single points of failures.
```
Comment (oskarth) [inline _posts/2020-04-16-wechat-replacement-need.md:216]:
```suggestion
**Decentralization**. Cryptocurrencies, projects like libp2p and IPFS. Need to be mindful here of many projects that claim decentralization but are still vulnerable to single points of failures, such as gateways.
```

File: _posts/2020-04-16-wechat-replacement-need.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+**Open source**. Big and only getting bigger. Including mainstream companies.
+
+## What's next?
+
+We've look at what WeChat provides and what we'd like an alternative to look like. We've also seen a few principal modules that are necessary to achieve those goals. To achieve all of this is a daunting task, and one might call it overly ambitiuous. We've also seen how far we've come with some of the goals, and how a lot of the pieces are there, in one form or another. Then it is a question of putting them all together in the right mix.
+
+The good news is that a lot of people are working all these building blocks and thinking about these problems. Compared to a few years ago we've come quite far when it comes to p2p infrastructure, privacy, security, scalability, and general developer mass and mindshare. If you want to join us in building some of these building blocks, and assembling them, check out our forum.
+
+PS. We are hiring. DS
```
Comment (oskarth) [inline _posts/2020-04-16-wechat-replacement-need.md:242]:
```suggestion
PS. We are [hiring protocol engineers](https://status.im/our_team/open_positions.html). DS
```

File: _posts/2020-04-16-wechat-replacement-need.md (hunk 1 above)
Comment (oskarth) [inline _posts/2020-04-16-wechat-replacement-need.md:244]:
```suggestion
```

File: _posts/2020-04-16-wechat-replacement-need.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+
+Even so, if the system is closed source we don't know what it is doing. Perhaps the app communicating is also uploading data to another place, or somehow making it possible to see who is who and act accordingly.
+
+You might notice that just one simple property, self-sovereign identity, leads to a slew of other requirements and properties. You might also notice that WeChat is far from alone in this, even if their identity requirements might be a bit stringent than, say, Telegram. Their control aspects are also a bit more extreme, at least for someone with western sensibilities [2].
+
+Most user facing applications have similar issues, Google Apps/FB/Twitter etc. For popular tools that have this built in, we can look at git - which is truly decentralized and have keypair at the bottom. It is for a very specific technical domain, and even then people rely on Github. Key management is fairly difficult even for technical people, and for normal people even more so. Banks are generally far behind on this tech, relying on arcane procedures and special purpose hardware for 2FA. That's another big issue.
+
+Let's shift gears a bit and talk about more  some other functional requirements.
```
Comment (oskarth) [inline _posts/2020-04-16-wechat-replacement-need.md:138]:
```suggestion
Let's shift gears a bit and talk about some other functional requirements.
```

File: _posts/2020-04-16-wechat-replacement-need.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+
+## Where are we now?
+
+Let's look back at some of desirable properties we set out in the beginning and see how close we are to building out the necessary components. Is it realistic at all or just a pipe dream? We'll see that there are many building blocks in place, and there's reason for hope.
+
+**Self-sovereignity identity**. Public key crypto and web of trust like constructs makes this possible.
+
+**Pseudonymity, and ideally total anonymity**. Pseudonymity can largely be achieved with public key crypto. For transactions, pseudonymity exists in systems in most cryptocurrencies. The challenge is linkage across time, especially when interfacing with other "legacy" system. There are stronger constructs that are actively being worked on and are promising here, such as mixnets (Nym), mixers (Wasabi Wallet, Tornado.Cash) and zero knowledge proofs (Zcash, Ethereum, Starkware). This area of applied research has exploded over the last few years.
```
Comment (oskarth) [inline _posts/2020-04-16-wechat-replacement-need.md:210]:
```suggestion
**Pseudonymity, and ideally total anonymity**. Pseudonymity can largely be achieved with public key crypto and open systems that allow for permissionless participation. For transactions, pseudonymity exists in systems in most cryptocurrencies. The challenge is linkage across time, especially when interfacing with other "legacy" system. There are stronger constructs that are actively being worked on and are promising here, such as mixnets (Nym), mixers (Wasabi Wallet, Tornado.Cash) and zero knowledge proofs (Zcash, Ethereum, Starkware). This area of applied research has exploded over the last few years.
```

**Comment by decanus** [comment]
@oskarth I believe you can probably format the footnotes in kramdown style. See how I did it here: https://github.com/decanus/dean.eigenmann.me/blob/master/_posts/2020-02-17-cap-theorem.md

And then how it renders: https://dean.eigenmann.me/blog/2020/02/17/cap-theorem/

**Comment by oskarth** [comment]
Thanks, added cite and image

Will post my tonight

**Pull request merged**